modules/component/src/listComponentState.ts

```diff
@@ -0,0 +1,44 @@
+import {Either, List} from 'standard-data-structures'
+
+type ListKey = string | number
+
+export class ListComponentState<S> {
+  private constructor(
+    private baseInit: () => S,
+    private readonly items: List<ListKey> = List.empty(),
+    private readonly lookUp: {[k in ListKey]?: S} = {}
```

Comment:
make `items` and `lookUp` as required params and send them via `ListComponentState.of(baseInit)`.

`ListComponentState.of` should not take any optional params. Internally while creating a clone use `new ListComponentState` and pass on items and lookUp.